tests/api/ocsp.c

```diff
@@ -0,0 +1,539 @@
+    ExpectIntEQ(ret, WOLFSSL_FAILURE);
+    response->sig[0] ^= 0xff;
+
+    /* cert embedded and in certs, no store needed bc OCSP_TRUSTOTHER */
+    ret = wolfSSL_OCSP_basic_verify(response, certs, NULL, OCSP_TRUSTOTHER);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    /* this should also pass */
+    ret = wolfSSL_OCSP_basic_verify(response, certs, store, OCSP_NOINTERN);
+    ;
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    /* this should not */
+    ret = wolfSSL_OCSP_basic_verify(response, NULL, store, OCSP_NOINTERN);
+    ;
+    ExpectIntNE(ret, WOLFSSL_SUCCESS);
+    wolfSSL_OCSP_RESPONSE_free(response);
+
+    /* cert not embedded, not certs */
+    ptr = (const unsigned char*)resp_nocert;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp_nocert));
+    ExpectPtrNE(response, NULL);
+    ret = wolfSSL_OCSP_basic_verify(response, NULL, store, 0);
+    ExpectIntNE(ret, WOLFSSL_SUCCESS);
+    wolfSSL_OCSP_RESPONSE_free(response);
+
+    wolfSSL_sk_X509_pop_free(certs, wolfSSL_X509_free);
+    wolfSSL_X509_STORE_free(store);
+
+    ret = test_ocsp_create_x509store(&store, root_ca_cert_pem,
+        sizeof(root_ca_cert_pem));
+    ExpectIntEQ(ret, TEST_SUCCESS);
+    ret = test_create_stack_of_x509(&certs, root_ca_cert_pem,
+        sizeof(root_ca_cert_pem));
+    ExpectIntEQ(ret, TEST_SUCCESS);
+
+    /* multiple responses in a ocsp response */
+    ptr = (const unsigned char*)resp_multi;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp_multi));
+    ExpectPtrNE(response, NULL);
+    ret = wolfSSL_OCSP_basic_verify(response, certs, store, 0);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    wolfSSL_OCSP_RESPONSE_free(response);
+
+    /* cert in certs, cert verified on store, not authorized to verify all
+     * responses */
+    ptr = (const unsigned char*)resp_bad_noauth;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp_bad_noauth));
+    ExpectPtrNE(response, NULL);
+    ret = wolfSSL_OCSP_basic_verify(response, certs, store, 0);
+#ifndef WOLFSSL_NO_OCSP_ISSUER_CHECK
+    ExpectIntEQ(ret, WOLFSSL_FAILURE);
+#else
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+#endif
+    /* should pass with OCSP_NOCHECKS ...*/
+    ret = wolfSSL_OCSP_basic_verify(response, certs, store, OCSP_NOCHECKS);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    /* or with OSCP_TRUSTOTHER */
+    ret = wolfSSL_OCSP_basic_verify(response, certs, store, OCSP_TRUSTOTHER);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    wolfSSL_OCSP_RESPONSE_free(response);
+
+    wolfSSL_sk_X509_pop_free(certs, wolfSSL_X509_free);
+    wolfSSL_X509_STORE_free(store);
+
+    return EXPECT_RESULT();
+}
+#else
+int test_ocsp_basic_verify(void) { return TEST_SKIPPED; }
+#endif /* HAVE_OCSP  && (OPENSSL_ALL || OPENSSL_EXTRA) */
+
+#if defined(HAVE_OCSP) && defined(HAVE_MANUAL_MEMIO_TESTS_DEPENDENCIES) &&     \
+    defined(HAVE_CERTIFICATE_STATUS_REQUEST) && !defined(WOLFSSL_NO_TLS12) &&  \
+    (defined(OPENSSL_ALL) || defined(OPENSSL_EXTRA))
```

Comment:
How is this ever run? `HAVE_MANUAL_MEMIO_TESTS_DEPENDENCIES` only exists in `api.c`.

Reply:
Update the `#endif` comment.